WordPress/src/main/java/org/wordpress/android/ui/reader/repository/ReaderPostRepository.kt

```diff
@@ -3,451 +3,182 @@ package org.wordpress.android.ui.reader.repository
-                "        \"type\": \"image\"\n" +
-                "      },\n" +
-                "      \"use_excerpt\": false,\n" +
-                "      \"is_following_conversation\": false\n" +
-                "    }\n" +
-                "  ],\n" +
-                "  \"meta\": {\n" +
-                "    \"links\": {\n" +
-                "      \"counts\": \"https:\\/\\/public-api.wordpress.com\\/rest\\/v1.2\\/" +
-                "sites\\/53424024\\/post-counts\\/post\"\n" +
-                "    },\n" +
-                "    \"next_page\": \"value=2020-06-02T10%3A00%3A08-04%3A00&id=42190\",\n" +
-                "    \"wpcom\": true\n" +
-                "  }\n" +
-                "}"
+class ReaderPostRepository(
+    private val bgDispatcher: CoroutineDispatcher,
+    private val readerTag: ReaderTag,
+    private val getPostsForTagUseCase: GetPostsForTagUseCase,
+    private val getNumPostsForTagUseCase: GetNumPostsForTagUseCase,
+    private val shouldAutoUpdateTagUseCase: ShouldAutoUpdateTagUseCase,
+    private val getPostsForTagWithCountUseCase: GetPostsForTagWithCountUseCase,
+    private val fetchPostsForTagUseCase: FetchPostsForTagUseCase,
+    private val readerUpdatePostsEndedHandler: ReaderUpdatePostsEndedHandler,
+    private val postLikeActionUseCase: PostLikeActionUseCase
+) : CoroutineScope {
+    override val coroutineContext: CoroutineContext
+        get() = bgDispatcher
+
+    private var isStarted = false
+
+    private val _mutablePosts = ReactiveMutableLiveData<ReaderPostList>(
+            onActive = { onActivePosts() }, onInactive = { onInactivePosts() })
+    val posts: LiveData<ReaderPostList> = _mutablePosts
+
+    private val _communicationChannel = MutableLiveData<Event<ReaderRepositoryCommunication>>()
+    val communicationChannel: LiveData<Event<ReaderRepositoryCommunication>> = _communicationChannel
+
+    fun start() {
+        if (isStarted) return
+
+        isStarted = true
+        readerUpdatePostsEndedHandler.start(readerTag,
+                ReaderUpdatePostsEndedHandler.setUpdatePostsEndedListeners(
+                        this::onNewPosts,
+                        this::onChangedPosts, this::onUnchanged, this::onFailed
+                ))
     }
 
-    private val mutableDiscoveryFeed = MutableLiveData<ReaderPostList>()
-    val discoveryFeed: LiveData<ReaderPostList> = mutableDiscoveryFeed
+    fun stop() {
+        getPostsForTagUseCase.stop()
+        getNumPostsForTagUseCase.stop()
+        shouldAutoUpdateTagUseCase.stop()
+        getPostsForTagWithCountUseCase.stop()
+        readerUpdatePostsEndedHandler.stop()
+    }
+
+    fun getTag(): ReaderTag {
+        return readerTag
+    }
 
-    suspend fun getDiscoveryFeed(): LiveData<ReaderPostList> =
-            withContext(bgDispatcher) {
-                delay(TimeUnit.SECONDS.toMillis(5))
-                getMockDiscoverFeed()
+    fun performLikeAction(post: ReaderPost, isAskingToLike: Boolean, wpComUserId: Long) {
```

Comment:
My gut says that there is an opportunity to move this into an "PostLikeManager" of sorts. Keep the repository lighter and the logic SRPish.

Reply:
I agree that we might want to consider extracting this logic outside of this class. Having said that, I'd personally leave it here for now and extract it when we have all the actions implemented and we see what could be extracted. Wdyt?

Reply:
> leave it here for now
👍 

Yes, I can live with the todo's as placeholder reminders during development.